Allow algorithm 35 with v4 keys (#174)
openpgp-pqc/draft-openpgp-pqc: draft-ietf-openpgp-pqc.md

Furthermore, when performing the explicitly listed operations in {{x25519-kem}} or {{x448-kem}} it is REQUIRED to follow the specification and security advisory mandated from the respective elliptic curve specification.

## Key version binding

> These asymmetric algorithms are to be used only in v6 TPKs and TSKs, with the single exception of ML-KEM-768+X25519 (algorithm ID 35).

Reviewer: ```suggestion
These asymmetric algorithms are to be used only in v6 (and newer) keys and certificates, with the single exception of ML-KEM-768+X25519 (algorithm ID 35).
```

Author: ```suggestion
All (PQ/T) asymmetric algorithms are to be used only in v6 TPKs and TSKs, with the single exception of ML-KEM-768+X25519 (algorithm ID 35).
```

> Algorithm 35 is also permitted for use in v4 TPKs and TSKs for the purpose of defending OpenPGP messages encrypted to existing certificates against store-now, decrypt-later attacks that use quantum computing machinery.

Reviewer: ```suggestion
For existing certificates, algorithm 35 is also permitted for use in v4 encryption-capable subkeys.
This permits the keyholder of an existing v4 certificate to add such a subkey to defend against store-now, decrypt-later attacks from quantum computers without moving to a new primary key.
New keys and certificates should use v6 for the full range of post-quantum algorithms. 
```

Furthermore, when performing the explicitly listed operations in {{x25519-kem}} or {{x448-kem}} it is REQUIRED to follow the specification and security advisory mandated from the respective elliptic curve specification.

## Key version binding

> All (PQ/T) asymmetric algorithms are to be used only in v6 (and newer) keys and certificates, with the single exception of ML-KEM-768+X25519 (algorithm ID 35).
> For existing certificates, algorithm 35 is also permitted for use in v4 encryption-capable subkeys.
> This permits the keyholder of an existing v4 certificate to add such a subkey to defend against store-now, decrypt-later attacks from quantum computers without moving to a new primary key.

Author: ```suggestion
All (PQ/T) asymmetric algorithms are to be used only in v6 (and newer) keys and certificates, with the single exception of ML-KEM-768+X25519 (algorithm ID 35), which is also allowed in v4 encryption-capable subkeys.
This permits the keyholder of an existing v4 certificate to add such a subkey to defend against store-now, decrypt-later attacks from quantum computers without moving to a new primary key.
```

> New keys and certificates should use v6 for the full range of post-quantum algorithms.

Author: ```suggestion
```

### Key Material Packets {#mlkem-ecc-key}

The composite ML-KEM-768 + X25519 (algorithm ID 35) MUST be used only with v4 or v6 keys, as defined in [RFC9580], or newer versions defined by updates of that document.

> The composite ML-KEM-1024 + X448 (algorithm ID 36) MUST be used only with v6 keys, as defined in [RFC9580], or newer versions defined by updates of that document

Reviewer: This seems clunky, but i think the text is correct.

If you want to be friendlier, you could also put a note at the top of the draft (in the introduction?) that says something like "these asymmetric algorithms should be used only in v6 TPKs and TSKs, with the single exception of algorithm ID 35.  Algo 35 is also permitted for use in v4 TPKs and TSKs for the purpose of defending OpenPGP messages encrypted to existing certificates against store-now, decrypt-later attacks that use quantum computing machinery."

Author: I pushed a version of this suggestion. We'll discuss it tomorrow morning

### Key Material Packets {#mlkem-ecc-key}

The composite ML-KEM-768 + X25519 (algorithm ID 35) MUST be used only with v4 or v6 keys, as defined in [RFC9580], or newer versions defined by updates of that document.

> The composite ML-KEM-1024 + X448 (algorithm ID 36) MUST be used only with v6 keys, as defined in [RFC9580], or newer versions defined by updates of that document

Author: ```suggestion
The composite ML-KEM-1024 + X448 (algorithm ID 36) MUST be used only with v6 keys, as defined in [RFC9580], or newer versions defined by updates of that document.
```
